Plugin~/WebRTCPlugin/DummyAudioDevice.cpp

```diff
@@ -38,18 +37,46 @@ namespace webrtc
                 int64_t ntp_time_ms = -1;
                 char data[kBytesPerSample * kChannels * kSamplesPerFrame];
 
-                audio_transport_->PullRenderData(kBytesPerSample * 8, kSamplingRate,
+                (*audio_transport_).PullRenderData(kBytesPerSample * 8, kSamplingRate,
                     kChannels, kSamplesPerFrame, data,
                     &elapsed_time_ms, &ntp_time_ms);
             }
         }
 
         int64_t deltaTimeMillis = rtc::TimeMillis() - currentTime;
         if (deltaTimeMillis < 10) {
-            SleepMs(10 - deltaTimeMillis);
+            SleepMs(10 - (deltaTimeMillis + 1));
         }
         return true;
     }
 
+    void DummyAudioDevice::RecordingThread() {
+        bool buffer_init = false;
```

Comment:
Is this value needed?

Reply:
Yes, `buffer_init` prevents too many audio buffering cause latency.

Reply:
No, I forgot to delete it. It has been removed from e9bbdec